src/Parser.ts

```diff
@@ -94,20 +106,31 @@ export class Parser {
   }
 
   consume(ttype: TokenType) {
-    this.advance()
+    while (!this.isAtEnd()) {
+      if (this.check(ttype)) {
+        return this.advance()
+      }
+      throw new ParserError('unexpected token')
+    }
+    throw new ParserError('unexpected end of input')
   }
 
   peek(): Token {
     return this.tokens[this.index]
   }
 
   advance(): Token {
-    this.index++
+    if (!this.isAtEnd()) this.index++
     return this.previous()
   }
 
   previous(): Token {
-    return this.tokens[this.index]
+    if (this.tokens[this.index - 1] == undefined) {
```

Comment:
let's make this more explicit -

```
if (this.index < 1) { ...
```

Reply:
Simplified this logic to be more explicit.